docs: Add clarifications (#8344)
obsproject/obs-studio: docs/sphinx/reference-scenes.rst

.. function:: void obs_scene_enum_items(obs_scene_t *scene, bool (*callback)(obs_scene_t*, obs_sceneitem_t*, void*), void *param)

>    Enumerates scene items within a scene, starting with the bottommost scene
>    item, contrary to the Sources dock showing the topmost scene item as
>    the first, because rendering a scene starts with the bottommost scene item.

Reviewer: ```suggestion
   Enumerates scene items within a scene in order of the bottommost scene item
   to the topmost scene item.
```

While rewriting this, I began to think it was unnecessary to describe the Source dock behavior. As such, I've removed that portion. If you think it is actually needed, perhaps something like this, even though I'm not really happy with this either:
> This enumeration order is contrary to the Sources dock showing the topmost scene item as the first, because rendering a scene starts with the bottommost scene item.

Author: That is incredibly precise and indeed removes the need to describe the Source dock behavior! Thank you!
